### PR TITLE
Create Mindful Activities

### PR DESCRIPTION
**Pull request by michaelwehrley**
* Add form to record mindful activity
* Create `MindfulActivity` class and augment accordingly
* Add Mindful model specs
* Add Mindful provider specs
* Add small js to allow for selecting of activity type title

[#110760718]

### CODE REVIEW

File: app/models/content_providers/activities/mindful_activities/review_provider.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    module MindfulActivities
+      # Provides a form for a Participant to update previously planned
+      # mindful activities as well as log the emotions and
+      # additional notes or non compliance reasons.
+      class ReviewProvider < BitCore::ContentProvider
+        def render_current(options)
+          view_context = options.view_context
+
+          view_context.render(
+            template: "think_feel_do_engine/activities/"\
```
Comment (ericcf) [inline app/models/content_providers/activities/mindful_activities/review_provider.rb]: Why put these templates in the tfd_engine namespace? Are they overriding existing templates?

File: spec/models/content_providers/mindful_activities/index_provider_spec.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        let(:options) { double("options", view_context: view_context) }
+        let(:participant) { instance_double(Participant) }
+
+        before do
+          allow(participant)
+            .to receive_message_chain("activities.mindful_activities.order")
+            .and_return([])
+        end
+
+        it "#render_current" do
```
Comment (ericcf) [inline spec/models/content_providers/mindful_activities/index_provider_spec.rb]: Is this just meant to be a smoke test?

File: spec/models/content_providers/mindful_activities/new_provider_spec.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
+require "rails_helper"
```
Comment (ericcf) [inline spec/models/content_providers/mindful_activities/new_provider_spec.rb]: Do these content provider specs add test coverage beyond the functional spec?

File: app/models/content_providers/activities/mindful_activities/review_provider.rb (hunk 1 above)
Comment (michaelwehrley) [inline app/models/content_providers/activities/mindful_activities/review_provider.rb]: good catch.

File: spec/models/content_providers/mindful_activities/index_provider_spec.rb (hunk 2 above)
Comment (michaelwehrley) [inline spec/models/content_providers/mindful_activities/index_provider_spec.rb]: probably not that helpful.  I had written this before my feature spec, but I should be able to remove it.

File: spec/models/content_providers/mindful_activities/new_provider_spec.rb (hunk 3 above)
Comment (michaelwehrley) [inline spec/models/content_providers/mindful_activities/new_provider_spec.rb]: In these cases now - after the update - I do want to verify that these attributes and class name remain and are not edited later.

**Comment by michaelwehrley** [comment]
@ericcf thanks for the look. Let me know if you recommend any other changes.  Thanks!

**Comment by ericcf** [comment]
Thanks. :ship:

**Pull request merged**